client-core/src/service/wallet_state_service.rs

```diff
@@ -1,18 +1,18 @@
-use std::collections::BTreeMap;
-
 use parity_scale_codec::{Decode, Encode};
 use secstr::SecUtf8;
+use std::collections::BTreeMap;
 
-use chain_core::{
-    init::coin::Coin,
-    tx::data::{input::TxoPointer, output::TxOut, TxId},
-};
+use chain_core::tx::data::{input::TxoPointer, output::TxOut, TxId};
 use client_common::{Error, ErrorKind, Result, ResultExt, SecureStorage, Storage};
 
-use crate::types::TransactionChange;
+use crate::types::{TransactionChange, TransactionPending, WalletBalance};
+use chain_core::init::coin::{sum_coins, CoinError};
 
 /// key space of wallet state
 const KEYSPACE: &str = "core_wallet_state";
+/// If we can't find the transaction in the latest BLOCK_HEIGHT_ENSURE after it broadcasted,
+/// we'll rollback the pending transactions
+pub const BLOCK_HEIGHT_ENSURE: u64 = 50;
```

Comment:
I think it better to hardcoded as a standard,  users don't need to know the detailed logic about this, just wait some time. And it will make the chain's configure more complex if we make it configurable

Reply:
what if it's optionally configurable? If the argument isn't provided, it'll use the default one?

Reply:
Ok

Reply:
fixed